src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/AndroidComputeResPathsTests.cs

```diff
@@ -99,5 +111,212 @@ public void AndroidComputeResPathsTest (string identity, string logicalName, str
+				/* resource2 */ Path.Combine ("Resources", "layout", "layout1.axml"),
+			},
+		};
+
+		[Test]
+		[TestCaseSource (nameof (XA1030ConflictingResourceFilesSource))]
+		public void XA1030ConflictingResourceFiles (string resource1, string resource2)
+		{
+			var hash = Tools.Files.HashString (resource1 + resource2);
+			var projectDir = Path.Combine (Root, "temp", $"{nameof (XA1030ConflictingResourceFiles)}_{hash}");
+			TestOutputDirectories [TestContext.CurrentContext.Test.ID] = projectDir;
+
+			var path = Path.Combine (projectDir, resource1);
+			Directory.CreateDirectory (Path.GetDirectoryName (path));
+			File.WriteAllText (path, contents: "");
+			path = Path.Combine (projectDir, resource2);
+			Directory.CreateDirectory (Path.GetDirectoryName (path));
+			File.WriteAllText (path, contents: "");
+
+			var intermediateDir = Path.Combine (projectDir, "obj", "Debug", "res");
+			Directory.CreateDirectory (intermediateDir);
+
+			string previousWorking = Environment.CurrentDirectory;
+			try {
+				Environment.CurrentDirectory = projectDir;
+				var task = new AndroidComputeResPaths {
+					BuildEngine = engine,
+					ResourceFiles = new ITaskItem [] { new TaskItem (resource1), new TaskItem (resource2) },
+					IntermediateDir = intermediateDir,
+					LowercaseFilenames = true
+				};
+
+				Assert.IsFalse (task.Execute (), "Task should fail.");
+				Assert.IsTrue (errors.Count > 0, "Task should produce an error.");
+				BuildErrorEventArgs error = errors [0];
+				Assert.AreEqual ("XA1030", error.Code);
+				StringAssert.Contains (resource1, error.Message);
+				StringAssert.Contains (resource2, error.Message);
+			} finally {
+				Environment.CurrentDirectory = previousWorking;
+			}
+		}
+
+		static readonly object [] XA1031ConflictingFileLinksSource = new object [] {
+			new object [] {
				/* identity1 */ Path.Combine ("Resources", "layout", "layout1.xml"),
				/* identity2 */ Path.Combine ("Resources", "layout-xhdpi", "layout1.xml"),
				/* link1 */     "Resources\\layout\\layout2.xml",
				/* link2 */     "Resources\\layout-xhdpi\\layout1.xml",
				/* resources */ true,
			},
+			new object [] {
				/* identity1 */ Path.Combine ("Resources", "layout", "layout1.xml"),
				/* identity2 */ null,
				/* link1 */     "Resources\\layout\\layout2.xml",
				/* link2 */     "Resources\\layout\\Layout1.xml",
				/* resources */ true,
			},
+			new object [] {
				/* identity1 */ Path.Combine ("Resources", "layout", "layout1.xml"),
				/* identity2 */ Path.Combine ("Resources", "layout-xhdpi", "layout1.xml"),
				/* link1 */     "Resources\\layout\\layout2.xml",
				/* link2 */     "Resources\\layout-xhdpi\\Layout1.xml",
				/* resources */ true,
			},
+			new object [] {
				/* identity1 */ Path.Combine ("Assets", "asset1.txt"),
				/* identity2 */ Path.Combine ("Assets", "asset3.txt"),
				/* link1 */     "Assets\\asset2.txt",
				/* link2 */     "Assets\\Asset3.txt",
				/* resources */ false,
			},
```

Comment:
I see some weird character here:

![image](https://user-images.githubusercontent.com/840039/99263030-49da4480-27e4-11eb-8a75-4f8b4f5741d0.png)

Maybe that is on accident?

Reply:
Ha. That'll show me to try to get the newline character correct in a search and replace in Visual Studio on Windows. Let me fix that too.